theano/tensor/elemwise.py

```diff
@@ -1059,35 +1076,110 @@ def _c_all(self, node, nodename, inames, onames, sub):
                                      fortran=alloc_fortran)
             alloc += cgen.make_checks([list(range(nnested))], [odtype],
                                       dict(sub, lv0=oname))
-        olv_index = i  # index of the last output
+        # index of the last output
+        olv_index = i
+
+        # Array referencing all inputs
+        ilength = len(_inames)
+        inArray = """
+        PyArrayObject **inRef[%(ilength)s];
+        """ % locals()
+        for iname, j in izip(_inames, range(len(_inames))):
+            inArray += """
+            inRef[%(j)i]=&%(iname)s;
+            """ % locals()
+
+        # Array referencing all outputs
+        olength = len(_onames)
+        outArray = """
+        PyArrayObject **outRef[%(olength)s];
+        """ % locals()
+        for oname, j in izip(_onames, range(len(onames))):
+            outArray += """
+            outRef[%(j)i]=&%(oname)s;
+            """ % locals()
+
+        alloc += outArray + inArray
+
+        # Declaration and initialization of the hardcoded olv_index
+        # used by the code generated in elemwise_cgen
+        alloc += """
+        int olv_index;
+        olv_index = %(j)i;
+        """ % locals()
+
+        # If the node has a non empty destroy map
+        if(len(self.inplace_pattern.keys()) > 0):
```

Comment:
You can remove the .keys() call. len(a_dict) return the number of element in the dict.